test/unit/config/test_path_resolves_to.py

```diff
@@ -209,6 +209,15 @@ def test_kwargs_listify(mock_init, monkeypatch):
     assert config.path4 == ['my-config/new1', 'my-config/new2']
 
 
+def test_kwargs_as_list_listify(mock_init, monkeypatch):
+    # Expected: use values from kwargs; each value resolved and listified
+    new_path4 = ['new1', 'new2']
+    config = BaseAppConfiguration(path4=new_path4)
+
+    assert config._raw_config['path4'] == 'new1,new2'
```

Comment:
This surprised me a bit, I expected this to still be a list, but it is joined into a comma-separated string at https://github.com/ic4f/galaxy/blob/dev_yaml_list_format/lib/galaxy/config/__init__.py#L298 , but only for paths (`admin_users` doesn't have `resolves_to`).

Not sure if this can become a problem at some point, but I thought I should mention it.

Reply:
I saw it too; and now that you mention this, I think we shouldn't leave it as is: the issue here is that now the `strip_deprecated_dir` function has this unexpected side effect (for list values).  Which, apart from being bad design, can later lead to problems, of course.
I'll fix it. I won't be ideal, but it'll preserve the `value` data type. 
In general, I think the config module is becoming too complicated. I think, if we have a way to mark all path values in the schema (making `path-resolves_to` mandatory for paths is one way), the logic can be simplified considerably. I should get to it soon.

Reply:
@nsoranzo  Could you take a look at the latest commit?